library/src/main/java/me/grantland/widget/AutofitHelper.java

```diff
@@ -128,19 +131,49 @@ private static void autofit(TextView view, TextPaint paint, float minTextSize, f
                     displayMetrics);
         }
 
-        if (size < minTextSize) {
-            size = minTextSize;
+        if (mIsHeightFitting) {
+            int targetHeight = view.getHeight() - view.getPaddingTop() - view.getPaddingBottom();
+            if (targetHeight <= 0) {
+                if (size < minTextSize) {
+                    size = minTextSize;
+                }
+                view.setTextSize(TypedValue.COMPLEX_UNIT_PX, size);
+                return;
+            }
+
+            float textHeight = getTextHeight(text, paint, targetWidth, size);
+            float heightRatio = targetHeight / textHeight;
+            float newSize = size * heightRatio;
+            if (newSize < size) {
+                size = newSize;
+            }
```

Comment:
this method of calculating a fitting textSize to match the height is interesting and different than how we calculate fitting textSize to match the width... are we guaranteed that text height changes proportionally with textSize? if we are, could we use a similar method for width? if not, should we use a similar procedure?